tutorial/under-the-hood/Compilation.md

```diff
@@ -153,67 +161,70 @@ result as parameter.
 
 ```xml
 <x:scenario label="scenario">
-   <x:call function="f"/>
-   <x:expect label="expectations" test="predicate"/>
+   <x:call function="my:f"/>
+   <x:expect label="expectations" test="$x:result = 1"/>
 </x:scenario>
 ```
 
 ### Stylesheet
 
 ```xml
 <!-- generated from the x:scenario element -->
-<xsl:template name="x:d4e2">
-   <x:call function="f"/>
-   <xsl:variable name="x:result" as="item()*">
-      <xsl:sequence select="f()"/>
-   </xsl:variable>
+<xsl:template name="x:scenario1">
+  <xsl:message>scenario</xsl:message>
+  <x:scenario id="scenario1"
+              xspec=".../compilation-simple-suite.xspec">
+     <x:label>scenario</x:label>
+     <x:call>
+        <xsl:attribute name="function">my:f</xsl:attribute>
+     </x:call>
+     <xsl:variable name="x:result" as="item()*">
+        <xsl:sequence select="my:f()"/>
+     </xsl:variable>
    ... generate scenario data in the report ...
-   <xsl:call-template name="x:d4e4">
-      <xsl:with-param name="x:result" select="$x:result"/>
-   </xsl:call-template>
+     <xsl:call-template name="x:scenario1-expect1">
+        <xsl:with-param name="x:result" select="$x:result"/>
+     </xsl:call-template>
 </xsl:template>
 
 <!-- generated from the x:expect element -->
-<xsl:template name="x:d4e4">
+<xsl:template name="x:scenario1-expect1">
    <xsl:param name="x:result" required="yes"/>
-   <!-- expected result (none here) -->
-   <xsl:variable name="impl:expected" select="()"/>
+   <!-- expected result -->
+   <xsl:variable name="impl:expect-d6e4" select="()"/>
    <!-- wrap $x:result into a doc node if node()+ -->
```

Comment:
`if node()+` was a bug (fixed in #317). It should be `if possible`, `whenever possible` or something like that.
It would be nice if we could use this opportunity to correct it.

Reply:
Tried to fix this today, too, using "if possible".